sdk/containerregistry/azure-containerregistry/azure/containerregistry/aio/_async_container_registry_client.py

```diff
@@ -860,6 +871,92 @@ async def update_tag_properties(
+        :paramtype media_type: str
+        :returns: The digest of the set manifest, calculated by the registry.
+        :rtype: str
+        :raises ValueError: If the parameter repository or manifest is None,
+            or the digest in the response does not match the digest of the set manifest.
+        """
+        try:
+            if isinstance(manifest, MutableMapping):
+                data = _UnclosableBytesIO(json.dumps(manifest).encode())
+            else:
+                data = _UnclosableBytesIO(manifest.read())
+            tag_or_digest = tag
+            if tag_or_digest is None:
+                tag_or_digest = _compute_digest(data)
+
+            response_headers = await self._client.container_registry.create_manifest(
+                name=repository,
+                reference=tag_or_digest,
+                payload=data,
+                content_type=media_type,
+                cls=_return_response_headers,
+                **kwargs
+            )
+            digest = response_headers['Docker-Content-Digest']
+            if not _validate_digest(data, digest):
+                raise ValueError("The digest in the response does not match the digest of the set manifest.")
+        except Exception as e:
+            if repository is None or manifest is None:
+                raise ValueError("The parameter repository and manifest cannot be None.") from e
+            raise
+        return digest
+
+    @distributed_trace_async
+    async def get_manifest(self, repository: str, tag_or_digest: str, **kwargs) -> GetManifestResult:
+        """Get the manifest for an artifact.
+
+        :param str repository: Name of the repository.
+        :param str tag_or_digest: The tag or digest of the manifest to get.
+            When digest is provided, will use this digest to compare with the one calculated by the response payload.
+            When tag is provided, will use the digest in response headers to compare.
+        :returns: GetManifestResult
+        :rtype: ~azure.containerregistry.GetManifestResult
+        :raises ValueError: If the requested digest does not match the digest of the received manifest.
+        """
+        response = cast(
+            PipelineResponse,
+            await self._client.container_registry.get_manifest(
+                name=repository,
+                reference=tag_or_digest,
+                accept=SUPPORTED_MANIFEST_MEDIA_TYPES,
+                cls=_return_response,
+                **kwargs
+            )
+        )
+        media_type = response.http_response.headers['Content-Type']
+        manifest_bytes = await response.http_response.read()
+        manifest_json = response.http_response.json()
+        if tag_or_digest.startswith("sha256:"):
+            digest = tag_or_digest
+        else:
+            digest = response.http_response.headers['Docker-Content-Digest']
```

Comment:
What if the response header digest does not match the digest value passed in by the customer? Shouldn't we validate that?

Reply:
Yep, our plan for other languages was to do the same thing that @timovv had investigated the Docker CLI doing (described in [this issue](https://github.com/Azure/azure-sdk-for-net/issues/33778)).

Reply:
I think the logic here is correct, though -- in the first case, the passed-in digest is compared to the digest computed on the response body, and if not the header digest is compared to the response body.  If I'm understanding correctly, I think the only change needed to the current implementation is perhaps to use a different error message depending on the case you're in.

Reply:
> I think the logic here is correct, though -- in the first case, the passed-in digest is compared to the digest computed on the response body, and if not the header digest is compared to the response body. If I'm understanding correctly, I think the only change needed to the current implementation is perhaps to use a different error message depending on the case you're in.

Thanks for your feedback! Have updated the error message in my last commit.

Reply:
> What if the response header digest does not match the digest value passed in by the customer? Shouldn't we validate that?

No. Because: when digest value is passing, this digest value that customer provided should be correct. We will use it to validate the digest from the returned manifest, and use the customer provided digest(not the digest in response headers) in `GetManifestResult`.